PlayerModelReplacer.cs

```diff
@@ -390,49 +362,44 @@ public void UpdateMaterial()
 			}
 		}
 
-		public void ToggleAndBroadcast(bool playAudio = true)
+		public void ToggleAndBroadcast(bool playAudio)
 		{
             Debug.Log($"{PluginInfo.PLUGIN_GUID}: Toggling dog mode for you ({playerController.playerUsername})!");
             if (this.isDogActive)
 			{
-				this.EnableHumanModel();
+				this.EnableHumanModel(playAudio);
             }
 			else
 			{
-				this.EnableDogModel();
+				this.EnableDogModel(playAudio);
             }
 
-			// This is a hack in the meantime before the networking issue is fixed when another player joins an existing session
-            //if (Chainloader.PluginInfos.ContainsKey("me.swipez.melonloader.morecompany"))
-            //{
-            //    MoreCompanyPatch.HideCosmeticsForPlayer(playerController);
-            //}
-
             this.BroadcastSelectedModel(playAudio);
 		}
 
-        public void ReceiveBroadcastAndToggle(bool playAudio = true, bool isDog = false)
+        public void ReceiveBroadcastAndToggle(bool playAudio, bool isDog)
         {
             if (isDog)
             {
-                Debug.Log($"{PluginInfo.PLUGIN_GUID}:Turning ({playerController.playerUsername}) into a dog! Woof!");
-                this.EnableDogModel();
+                Debug.Log($"{PluginInfo.PLUGIN_GUID}: Turning {playerController.playerUsername} into a dog! Woof!");
+                this.EnableDogModel(playAudio);
                 if (Chainloader.PluginInfos.ContainsKey("me.swipez.melonloader.morecompany"))
                 {
                     MoreCompanyPatch.HideCosmeticsForPlayer(playerController);
                 }
             }
             else
             {
-                Debug.Log($"{PluginInfo.PLUGIN_GUID}:Turning ({playerController.playerUsername}) into a human! Damn!");
-                this.EnableHumanModel();
+                Debug.Log($"{PluginInfo.PLUGIN_GUID}:Turning {playerController.playerUsername} into a human!");
+                this.EnableHumanModel(playAudio);
 
                 if (Chainloader.PluginInfos.ContainsKey("me.swipez.melonloader.morecompany"))
                 {
 					if (playerController.IsOwner) // This should only be true once when you start up!
 					{
                         Debug.Log($"{PluginInfo.PLUGIN_GUID}: Hang on, you're {playerController.playerUsername}, we won't show your cosmetics!");
-						return;
+                        MoreCompanyPatch.HideCosmeticsForPlayer(playerController);
```

Comment:
Added this as some security, since I did notice a cosmetic bug when play testing with a larger lobby. This might be a grief though.